AgileCoffeeShop/Program.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+
+namespace AgileCoffeeShop
+{
+    class Program
+    {
+        #region Members
+        private static User user;
+        private static string urlFilePath;
+        private static List<CoffeeShop> coffeeShops;
+        private static Dictionary<string, double> distancesDictionary;
+        #endregion
+  
+        private static void ReadDataFromUrlFile(string urlFilePath)
+        {
+            WebClient client = new WebClient();
+            try
+            {
+                Stream stream = client.OpenRead(urlFilePath);
+                StreamReader reader = new StreamReader(stream);
+                string line = string.Empty;
+
+                while (!string.IsNullOrEmpty(line = reader.ReadLine()))
+                {
+                    string[] splitLine = line.Split(',');
+                    CoffeeShop coffeeShop = new CoffeeShop();
+                    coffeeShop.Name = splitLine[0];
+                    ValidateCoordinate(splitLine[1]);
+                    coffeeShop.Coordinate.YCoordinate = Convert.ToDouble(splitLine[1]);
+                    ValidateCoordinate(splitLine[2]);
+                    coffeeShop.Coordinate.XCoordinate = Convert.ToDouble(splitLine[2]);
+                    coffeeShops.Add(coffeeShop);
+                }
+            }
+            catch(Exception exception)
+            {
+                Console.WriteLine("Invalid URL." + exception.Message + "Program will exit.\n Press any key to continue...");
+                Console.ReadKey();
+            }
+        }
+
+        public static void ValidateCoordinate (string coordinate)
+        {
+            double temp = 0;
+            try
+            {
+                temp = Convert.ToDouble(coordinate);
+            }
+            catch(Exception exception)
+            {
+                Console.WriteLine("Invalid coordinates." + exception.Message + "Program will exit.\nPress any key to continue...");
```

Comment:
Maybe you can store all the errors in a single place? Same for errors strings.

Reply:
Yes, I could store them in a separate class.